src/ui/controllers/booking.ts

```diff
@@ -181,19 +195,25 @@ export async function tryCancelBooking(
 	res: express.Response
 ) {
 	const course = req.course
+	const record = await learnerRecord.getCourseRecord(req.user, course)
+
+	if (!record.selectedDate) {
+		res.redirect('/')
```

Comment:
Is this where we want to redirect them?

Reply:
Ideally we'd send them some malware and take over their computer, but I don't think I'd get away with that. A user will only hit this if they're screwing around trying to do things they shouldn't so I'm not too bothered about the experience at the mo.